### PR TITLE
feat(codepipeline-actions): Add CAPABILITY_AUTO_EXPAND (#2851)

### LINKED ISSUE #2851: Add CAPABILITY_AUTO_EXPAND
Add CAPABILITY_AUTO_EXPAND to CloudFormationCapabilities to enable CodePipeline to deploy CloudFormation templates which include macros. Good example is AWS SAM generated ones.

### PR DESCRIPTION
**Pull request by ScOut3R**
BREAKING CHANGE: the capabilities property is now an array to support multiple capabilities.

Adds CAPABILITY_AUTO_EXPAND and support to define a list of capabilities for the CodePipeline action.

Closes #2851

### CODE REVIEW

File: packages/@aws-cdk/app-delivery/lib/pipeline-deploy-stack-action.ts [hunk 1]
```diff
@@ -166,13 +166,13 @@ export class PipelineDeployStackAction extends cdk.Construct {
   }
 }
 
-function cfnCapabilities(adminPermissions: boolean, capabilities?: cfn.CloudFormationCapabilities): cfn.CloudFormationCapabilities {
+function cfnCapabilities(adminPermissions: boolean, capabilities?: cfn.CloudFormationCapabilities[]): cfn.CloudFormationCapabilities[] {
   if (adminPermissions && capabilities === undefined) {
     // admin true default capability to NamedIAM
-    return cfn.CloudFormationCapabilities.NamedIAM;
+    return [cfn.CloudFormationCapabilities.NamedIAM];
```
Comment (RomainMuller) [inline packages/@aws-cdk/app-delivery/lib/pipeline-deploy-stack-action.ts]: I wonder if this should not contain `AutoExpand`, too.

File: packages/@aws-cdk/app-delivery/test/test.pipeline-deploy-stack-action.ts [hunk 2]
```diff
@@ -97,21 +97,21 @@ export = nodeunit.testCase({
...
-      capabilities: cfn.CloudFormationCapabilities.None,
+      capabilities: [cfn.CloudFormationCapabilities.None],
       adminPermissions: false,
     });
     new PipelineDeployStackAction(pipelineStack, 'DeployStack2', {
       stage: selfUpdateStage3,
       stack: stackWithAnonymousCapability,
       input: selfUpdatingStack.synthesizedApp,
-      capabilities: cfn.CloudFormationCapabilities.AnonymousIAM,
+      capabilities: [cfn.CloudFormationCapabilities.AnonymousIAM],
```
Comment (RomainMuller) [inline packages/@aws-cdk/app-delivery/test/test.pipeline-deploy-stack-action.ts]: Would like to have an additional test that uses `AutoExpand`.

File: packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts [hunk 3]
```diff
@@ -540,3 +540,14 @@ interface StatementTemplate {
 }
 
 type StatementCondition = { [op: string]: { [attribute: string]: string } };
+
+function parseCapabilities(capabilities: any): any {
```
Comment (RomainMuller) [inline packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts]: Can we avoid using `any` here? I would rather those be tightly typed as much as possible.

File: packages/@aws-cdk/app-delivery/lib/pipeline-deploy-stack-action.ts (hunk 1 above)
Comment (ScOut3R) [reply packages/@aws-cdk/app-delivery/lib/pipeline-deploy-stack-action.ts]: Yes, makes sense, I've added it. Please let me know if the changes are acceptable.

File: packages/@aws-cdk/app-delivery/test/test.pipeline-deploy-stack-action.ts (hunk 2 above)
Comment (ScOut3R) [reply packages/@aws-cdk/app-delivery/test/test.pipeline-deploy-stack-action.ts]: Added a couple more tests.

File: packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts (hunk 3 above)
Comment (ScOut3R) [reply packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts]: Yes, sorry for missing out on this one. I've updated the function.

**Comment by ScOut3R** [comment]
@RomainMuller sorry, I was delayed resolving the merge conflict. It should be alright now.

**Pull request merged**